chainerx_cc/chainerx/array_index.h

```diff
@@ -16,6 +16,14 @@ enum class ArrayIndexTag {
     kEllipsis,
 };
 
+enum class IndexBoundsMode {
+    // Index out-of-bounds handling modes for take
```

Comment:
TODO: remove this comment or update it with all the routines that support it
`choose` also allows index mode kwarg